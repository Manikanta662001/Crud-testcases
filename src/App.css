.storage{
  text-align: center;
}
table,th,td{
  margin: auto;
  border: 2px solid orchid;
  border-collapse:collapse;
}
th,td
{
  padding: 10px;
}
.App{
  background-color: antiquewhite;
}
div h1{
  color:purple;
}
button{
  background-color: rgb(12, 218, 190);
  color:rgb(207, 5, 190);
}