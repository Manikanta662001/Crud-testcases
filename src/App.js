import React from "react";
import Storage from "./components/Storage";
import Snapshot from "./components/Snapshot";
import TestElements from "./components/TestElements";
import "./App.css";
import TestEvents from "./components/TestEvents";
import TestAsync from "./components/TestAsync";
import Register from "./components/Register";
import Practice from "./components/Practice";
import Another from "./components/Another";
import Apicall from "./components/Apicall";
import List from "./components/List";
import TestWithMockData from "./components/TestwithMockdata";
import Api2 from "./components/Api2";
import AllEvents from "./components/AllEvents";
function App() {
  return (
    <div className="App">
      <h1>Hello</h1>
      <Storage />
      <Snapshot />
      <TestElements />
      <TestEvents />
      <TestAsync />
      <Register />
      <Practice />
      <Another />
      <Apicall url={"https://jsonplaceholder.typicode.com/posts"} />
      <List />
      <TestWithMockData data={[]}/>
      <Api2 />
      <AllEvents />
    </div>
  );
}

export default App;
