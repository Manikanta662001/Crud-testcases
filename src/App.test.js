import {render} from '@testing-library/react'
import App from './App'
describe('Testing App compo',()=>{
    it('Snapshot of App',()=>{
        const {asFragment} = render(<App/>)
        expect(asFragment(<App/>)).toMatchSnapshot()
    })
})
describe('practice testing',()=>{
    it('object assignmen',()=>{
        const data = {one:1}
        data['two'] = 2
        expect(data).toEqual({one:1,two:2})
    })
    it('null',()=>{
        const n = null
        const x=undefined;
        expect(n).toBeNull()
        expect(x).toBeUndefined()
    })
    test('two plus two', () => {
        const value = 2 + 2;
        expect(value).toBeGreaterThan(3);
        expect(value).toBeGreaterThanOrEqual(3.5);
        expect(value).toBeLessThan(5);
        expect(value).toBeLessThanOrEqual(4.5);
      
        // toBe and toEqual are equivalent for numbers
        expect(value).toBe(4);
        expect(value).toEqual(4);
      });
      test('adding floating point numbers', () => {
        const value = 0.1 + 0.2;
        // expect(value).toBe(0.3);           //This won't work because of rounding error
        expect(value).toBeCloseTo(0.3); // This works.
      });
      test('there is no I in team', () => {
        expect('team').not.toMatch(/I/);
      });
      test('but there is a "stop" in Christoph', () => {
        expect('Christoph').toMatch(/stop/);
      });

      const shoppingList = [
        'diapers',
        'kleenex',
        'trash bags',
        'paper towels',
        'milk',
      ];

      test('the shopping list has milk on it', () => {
        expect(shoppingList).toContain('milk')
      })
      

      const fetchData=()=>{
        return Promise.resolve('Peanut butter')
      }
      it('data fetching',async()=>{
        const data = await fetchData()
        expect(data).toBe('Peanut butter')
      })
})