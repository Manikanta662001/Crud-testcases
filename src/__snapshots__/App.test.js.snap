// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App compo Snapshot of App 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Hello
    </h1>
    <div
      class="storage"
    >
      <h1>
        Form
      </h1>
      <form>
        <label
          for="username"
        >
          Username: 
        </label>
        <input
          id="username"
          name="name"
          placeholder="Enter user name"
          type="text"
          value=""
        />
        <label>
          Age: 
        </label>
        <input
          name="age"
          placeholder="Enter user age"
          type="number"
          value="1"
        />
        <button>
          Submit
        </button>
      </form>
      <br />
      <hr />
      <input
        placeholder="search for users"
        type="search"
        value=""
      />
      <br />
      <hr />
      <div>
        <h3>
          Users list: 
        </h3>
        <div>
          No data available
        </div>
      </div>
    </div>
    <div>
      <h1>
        Snapshot
      </h1>
      <h1>
        Snap 2
      </h1>
      <h3>
        Hii this is Snapshot page
      </h3>
      <h4>
        Adding more content
      </h4>
      <h4>
        Adding more content
      </h4>
      <h6>
        Snapshot matching
      </h6>
    </div>
    <div>
      <h1
        data-testid="count"
      >
        0
      </h1>
      <button>
        Increment
      </button>
      <button
        disabled=""
      >
        Decrement
      </button>
    </div>
    <div>
      <h1
        data-testid="count"
      >
        0
      </h1>
      <button
        data-testid="Arrow-up"
      >
        Increment
      </button>
      <button
        data-testid="Arrow-down"
      >
        Decrement
      </button>
    </div>
    <div>
      <h1
        data-testid="count"
      >
        0
      </h1>
      <button
        data-testid="Arrow-up"
      >
        Increment
      </button>
      <button
        data-testid="Arrow-down"
        disabled=""
      >
        Decrement
      </button>
    </div>
    <div
      class="registration"
    >
      <div
        class="container"
      >
        <h2
          class="title"
        >
          Register
        </h2>
        <form
          class=""
        >
          <h6
            class="subtitle"
          >
            Please enter your details below to register yourself.
          </h6>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="name"
            >
              Name
            </label>
            <input
              class="form-control"
              id="name"
              name="name"
              placeholder="Enter your name"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="email"
            >
              Email
            </label>
            <input
              class="form-control"
              id="email"
              name="email"
              placeholder="Enter your email"
              type="email"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="password"
            >
              Password
            </label>
            <input
              class="form-control"
              id="password"
              name="password"
              placeholder="Enter your password"
              type="password"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              data-testid="selected_skill"
              for="password"
            >
              Select your skills
            </label>
            <div
              class=" css-b62m3t-container"
            >
              <span
                class="css-1f43avz-a11yText-A11yText"
                id="react-select-2-live-region"
              />
              <span
                aria-atomic="false"
                aria-live="polite"
                aria-relevant="additions text"
                class="css-1f43avz-a11yText-A11yText"
              />
              <div
                class=" css-1s2u09g-control"
              >
                <div
                  class=" css-319lph-ValueContainer"
                >
                  <div
                    class=" css-14el2xx-placeholder"
                    id="react-select-2-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    class=" css-6j8wv5-Input"
                    data-value=""
                  >
                    <input
                      aria-autocomplete="list"
                      aria-describedby="react-select-2-placeholder"
                      aria-expanded="false"
                      aria-haspopup="true"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      class=""
                      id="react-select-2-input"
                      role="combobox"
                      spellcheck="false"
                      style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class=" css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class=" css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="mb-3"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="newsletter"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="newsletter"
                title=""
              >
                Subscribe to our newsletter
              </label>
            </div>
          </div>
          <div>
            <button
              class="btn btn-secondary"
              type="submit"
            >
              Register
            </button>
            <button
              class="btn btn-primary"
              type="button"
            >
              Dummy
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      id="hai"
    >
      <div>
        Hello World
      </div>
      <button>
        Open dialog
      </button>
      <button>
        Close dialog
      </button>
      <div
        role="tablist"
      >
        <button
          aria-selected="true"
          role="tab"
        >
          Native
        </button>
        <button
          aria-selected="false"
          role="tab"
        >
          React
        </button>
        <button
          aria-selected="false"
          role="tab"
        >
          Cypress
        </button>
      </div>
      <section>
        <div
          aria-busy="false"
          role="alert"
        >
          Login failed
        </div>
        <div
          aria-busy="true"
          role="alert"
        >
          Error: Loading message...
        </div>
      </section>
      <input
        id="lastName"
        type="text"
      />
      <textarea
        id="messageTextArea"
      />
      <select>
        <option
          value=""
        >
          State
        </option>
        <option
          value="AL"
        >
          Alabama
        </option>
        <option
          value="AK"
        >
          Alaska
        </option>
        <option
          value="AZ"
        >
          Arizona
        </option>
      </select>
      <img
        alt="Incredibles 1 Poster"
      />
      <img
        alt="Incredibles 2 Poster"
      />
      <img
        alt="Incredibles 3 Poster"
      />
    </div>
    <div>
      <div>
        <label
          for="search"
        >
          Search :
        </label>
        <input
          id="search"
          type="text"
          value=""
        />
      </div>
      <p>
        Searches for ...
      </p>
    </div>
    <div>
      <button>
        Api call
      </button>
      <ul />
    </div>
    <div>
      <h1>
        List
      </h1>
      <div
        data-testid="record"
      >
        1 mani
      </div>
      <div
        data-testid="record"
      >
        2 vamsi
      </div>
      <div
        data-testid="record"
      >
        3 aneesh
      </div>
      <button>
        Add user
      </button>
      <button>
        Remove user
      </button>
    </div>
    <div>
      <h3>
         List 
      </h3>
      <button>
        Add to List
      </button>
      <button>
        Toggle text
      </button>
    </div>
    <h1>
      List of Users
    </h1>
    <p>
      No users found
    </p>
    <div
      style="text-align: center;"
    >
      <form>
        <h1>
          All Events
        </h1>
        <div>
          <label
            for="name"
          >
            Name: 
          </label>
          <input
            class="capital"
            id="name"
            name="name"
            placeholder="Enter Your Name"
            required=""
            type="text"
            value=""
          />
          <button>
            Change Boolean
          </button>
        </div>
        <div>
          <label
            for="location"
          >
            Location: 
          </label>
          <input
            id="location"
            name="location"
            placeholder="Enter Your Location"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="states"
          >
            Select State: 
          </label>
          <select
            id="states"
            name="states"
          >
            <option
              value=""
            />
            <option
              value="AP"
            >
              AP
            </option>
            <option
              value="TS"
            >
              TS
            </option>
            <option
              value="MH"
            >
              MH
            </option>
            <option
              value="NP"
            >
              NP
            </option>
            <option
              value="KA"
            >
              KA
            </option>
            <option
              value="TN"
            >
              TN
            </option>
          </select>
          <p />
        </div>
        <div>
          <label
            for="gender"
          >
            Gender: 
          </label>
          <div>
            <label>
              <input
                id="gender"
                name="gender"
                type="radio"
                value="Male"
              />
               Male
            </label>
            <label>
              <input
                id="gender"
                name="gender"
                type="radio"
                value="Female"
              />
               Female
            </label>
          </div>
        </div>
        <div>
          <button
            disabled=""
          >
            Submit
          </button>
        </div>
        <div>
          <label
            for="key"
          >
            keyUp & keyDown: 
          </label>
          <input
            placeholder="Enter Something....."
            type="text"
            value=""
          />
          <p>
            Key Pressed: 
          </p>
          <p>
            Key Released: 
          </p>
        </div>
      </form>
    </div>
    <div>
      <h1>
        Api Call with Fetch
      </h1>
      <div>
        Loading.....
      </div>
    </div>
  </div>
</DocumentFragment>
`;
